activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -205,7 +205,9 @@ def begin_transaction(options = {})
                                        run_commit_callbacks: run_commit_callbacks)
             end
 
-          transaction.materialize! unless @connection.supports_lazy_transactions? && lazy_transactions_enabled?
+          if options[:lazy] == false || !(@connection.supports_lazy_transactions? && lazy_transactions_enabled?)
```

Comment:
I think this might be one of those rare cases where a negative condition inside an `unless` actually reads better?

```suggestion
          unless @connection.supports_lazy_transactions? && lazy_transactions_enabled? && options[:lazy] != false
```